src/mngt/ocf_mngt_cache.c

```diff
@@ -1237,7 +1234,13 @@ static int _ocf_mngt_cache_start(ocf_ctx_t ctx, ocf_cache_t *cache,
 
 	ocf_cache_log(tmp_cache, log_debug, "Metadata initialized\n");
 
-	_ocf_mngt_cache_init(tmp_cache, &params);
+	INIT_LIST_HEAD(&tmp_cache->io_queues);
+
+	/* Init Partitions */
+	ocf_part_init(tmp_cache);
+
+	if (!tmp_cache->metadata.loaded)
+		_ocf_mngt_cache_init(tmp_cache, &params);
```

Comment:
I think some pieces of initialization from "ocf_part_init are missing when loading from shmem:
1. is cache->lst_part filled?
2. are partition reference freezed (ocf_refcnt_freeze(&cache->user_parts[i_part].cleaning.counter)) ?